src/fe-small-advertisements/src/components/App.tsx

```diff
@@ -0,0 +1,32 @@
+import React, { ChangeEvent, FormEvent, useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { changeNameActionCreator } from '../actions';
+import { RootState } from '../reducers';
+
+function App() {
+  const { name } = useSelector((state: RootState) => state.user);
+  const dispatch = useDispatch();
+  const [newName, setNewName] = useState('');
+
+  const handleNewName = (e: ChangeEvent<HTMLInputElement>): void => {
+    setNewName(e.target.value);
+  };
+
+  const handleUpdateTitle = (e: FormEvent<HTMLFormElement>): void => {
+    e.preventDefault();
+    dispatch(changeNameActionCreator({ name: newName }));
+    setNewName('');
+  };
+
+  return (
+    <>
+      <p>My name is: {name}</p>
+      <form onSubmit={handleUpdateTitle}>
+        <input value={newName} onChange={handleNewName} />
+        <button type="submit">change name</button>
+      </form>
+    </>
+  );
+}
+
+export default App;
```

Comment:
I think we agreed these will be `const`, mind swapping it?

Reply:
Right, we are never going to have default exports for the components, right? I have being reading about the advantages of it